### ISSUE TITLE
False positive error reported referencing protocols in cljs.core namespace

### ISSUE THREAD

**Issue by weavejester** [clj-kondo/clj-kondo#1535]
**version** v2022.01.13

**platform** MacOS 11.4

**problem**

An false positive error is reported on ClojureScript files that reference protocols in the `cljs.core` namespace.

**repro**

```bash
$ echo '(defn editable? [coll] (satisfies? cljs.core.IEditableCollection coll))' > test.cljs
$ clj-kondo --lint test.cljs
test.cljs:1:36: error: Unresolved symbol: cljs
linting took 14ms, errors: 1, warnings: 0
```

**expected behavior**

No error should be reported, as the `cljs.core` namespace is implicitly loaded in ClojureScript, along with the protocols in it.

**Comment by borkdude** [comment]
I would say writing `ns/Protocol` is more idiotmatic than `ns.Protocol` anwyay.

**Comment by borkdude** [comment]
Also given how it works in CLJ:

``` clojure
user=> (ns foo)
nil
foo=> (defprotocol Foo)
Foo
foo=> (ns bar)
nil
bar=> (satisfies? foo/Foo 1)
false
bar=> (satisfies? foo.Foo 1)
Execution error (NullPointerException) at bar/eval169 (REPL:1).
null
```

**Comment by weavejester** [comment]
Ah, thanks. That's probably a better solution.

**Comment by weavejester** [comment]
For completeness, the same thing happens for types (e.g. `cljs.core.PersistentQueue.EMPTY`), but this also looks fixable by using `/` instead of `.` in the namespace divider (e.g. `cljs.core/PersistentQueue.EMPTY`).

**Comment by stevenproctor** [comment]
👍  
I just found this issue looking to fix the same thing that @weavejester just called out regarding `PersistentQueue` in cljs, and saw it update before my eyes.

**Comment by borkdude** [comment]
Thanks, will fix on the next version!

**Comment by borkdude** [comment]
In addition to supporting this, should we also emit an optional warning to suggest `cljs.core/Var` in preference over `cljs.core.Var`? Even `core.cljs` seems to do this, so maybe not.

**Comment by borkdude** [comment]
Should be fixed on master. Please test your CLJS code-bases to see if this is fixed and to check regressions.
